airflow/utils/dag_processing.py

```diff
@@ -63,204 +62,109 @@ class SimpleDag(BaseDag):
 
 
-class SimpleTaskInstance:
-    def __init__(self, ti):
-        self._dag_id = ti.dag_id
-        self._task_id = ti.task_id
-        self._execution_date = ti.execution_date
-        self._start_date = ti.start_date
-        self._end_date = ti.end_date
-        self._try_number = ti.try_number
-        self._state = ti.state
-        self._executor_config = ti.executor_config
-        if hasattr(ti, 'run_as_user'):
-            self._run_as_user = ti.run_as_user
-        else:
-            self._run_as_user = None
-        if hasattr(ti, 'pool'):
-            self._pool = ti.pool
-        else:
-            self._pool = None
-        if hasattr(ti, 'priority_weight'):
-            self._priority_weight = ti.priority_weight
-        else:
-            self._priority_weight = None
-        self._queue = ti.queue
-        self._key = ti.key
-
-    @property
-    def dag_id(self):
-        return self._dag_id
-
-    @property
-    def task_id(self):
-        return self._task_id
-
-    @property
-    def execution_date(self):
-        return self._execution_date
-
-    @property
-    def start_date(self):
-        return self._start_date
-
-    @property
-    def end_date(self):
-        return self._end_date
-
-    @property
-    def try_number(self):
-        return self._try_number
-
-    @property
-    def state(self):
-        return self._state
-
-    @property
-    def pool(self):
-        return self._pool
-
-    @property
-    def priority_weight(self):
-        return self._priority_weight
-
-    @property
-    def queue(self):
-        return self._queue
-
-    @property
-    def key(self):
-        return self._key
-
-    @property
-    def executor_config(self):
-        return self._executor_config
-
-    @provide_session
-    def construct_task_instance(self, session=None, lock_for_update=False):
-        """
-        Construct a TaskInstance from the database based on the primary key
-
-        :param session: DB session.
-        :param lock_for_update: if True, indicates that the database should
-            lock the TaskInstance (issuing a FOR UPDATE clause) until the
-            session is committed.
-        """
-        TI = airflow.models.TaskInstance
-
-        qry = session.query(TI).filter(
-            TI.dag_id == self._dag_id,
-            TI.task_id == self._task_id,
-            TI.execution_date == self._execution_date)
-
-        if lock_for_update:
-            ti = qry.with_for_update().first()
-        else:
-            ti = qry.first()
-        return ti
-
-
 class SimpleDagBag(BaseDagBag):
     """
     A collection of SimpleDag objects with some convenience methods.
     """
 
-    def __init__(self, simple_dags):
+    def __init__(self, simple_dags: List[SimpleDag]):
         """
         Constructor.
 
         :param simple_dags: SimpleDag objects that should be in this
-        :type list(airflow.utils.dag_processing.SimpleDagBag)
+        :type List(airflow.utils.dag_processing.SimpleDag)
```

Comment:
```suggestion
```

Do we really need this since we already define it in the function header?

Reply:
Nope. I added sphinx-autodoc-typehint that will add those automatically. I will review all the types in the affected files and see if there are any :type left.

Reply:
Won't work with autoapi :(: https://github.com/readthedocs/sphinx-autoapi/issues/139. Tried even latest version of autoapi, but it does not work.